### PR TITLE
Add sok tests in Merlin

### PR DESCRIPTION
**Pull request by EmmaQiaoCh**
* change dockerfile to build sok unit test;
* keep the source folder which has sok test scripts;
* add sok test in ci/test_container.sh

### CODE REVIEW

**Comment by benfred** [comment]
This looks good to me - have you built the container and tested that all the tests pass @EmmaQiaoCh  ?

**Comment by jperez999** [comment]
The following line needs to be removed https://github.com/NVIDIA-Merlin/Merlin/blob/main/docker/training/dockerfile.tf#L116 and you need to figure out another way to get your package recognized. If you like that line in the dockerfile, anything you pip install after the ENV VAR is updated will fail, which is why we originally did https://github.com/NVIDIA-Merlin/Merlin/pull/317/. If you want to see an issue I hit, install jupyterlab and try to run... tensorboard will fail to load. Without the added path, it loads correctly.

**Comment by EmmaQiaoCh** [comment]
> The following line needs to be removed https://github.com/NVIDIA-Merlin/Merlin/blob/main/docker/training/dockerfile.tf#L116 and you need to figure out another way to get your package recognized. If you keep that line in the dockerfile, anything you pip install after the ENV VAR is updated will fail, which is why we originally did #317. If you want to see an issue I hit, install jupyterlab and try to run... tensorboard plugin will fail to load. Without the added path, it loads correctly.

Hi Julio,
Thanks for your comments, I can reproduce the jupyterlab problem. I tried to downgrade setuptools to '59.4.0', sok installed normally and no ENV needs. And the following distributed_embeddings also installed successfully. But still met the tensorboard plugin load failure. It works on @benfred 's PR, but sok test failed, so we need more time to fix these problems.

**Comment by EmmaQiaoCh** [comment]
I've removed env PYTHONPATH and change to use the pip to install SOK, Please review the code changes. Thanks
Please Note: these changes depends on https://github.com/NVIDIA-Merlin/HugeCTR/pull/317 and https://github.com/NVIDIA-Merlin/HugeCTR/pull/320

File: docker/training/dockerfile.tf [hunk 1]
```diff
@@ -83,7 +83,8 @@ RUN git clone https://github.com/NVIDIA-Merlin/Merlin/ /Merlin
 # Install HugeCTR
 ENV LD_LIBRARY_PATH=/usr/local/hugectr/lib:$LD_LIBRARY_PATH \
     LIBRARY_PATH=/usr/local/hugectr/lib:$LIBRARY_PATH \
-    PYTHONPATH=/usr/local/hugectr/lib:$PYTHONPATH
+    PYTHONPATH=/usr/local/hugectr/lib:$PYTHONPATH \
```
Comment (benfred) [inline docker/training/dockerfile.tf:86]:
```suggestion
```

**Review by benfred — CHANGES_REQUESTED** [review]

**Pull request closed without merging**